### DISCUSSION TITLE
Semantic Release

### DISCUSSION THREAD

**Discussion by samuelcastro** [UnlyEd/next-right-now#284] [General]
Have you considered [Semantic Release](https://github.com/semantic-release/semantic-release)?

I use it in another project along with with [commitizen](https://github.com/commitizen/cz-cli) and the dev experience is really great, but the simple fact it detects the next release (minor/major/breaking changes) automatically based on the commits.

**Comment by Vadorequest** [comment]
It's similar to what we use at the moment: https://github.com/UnlyEd/next-right-now/blob/v2-mst-aptd-at-lcz-sty/.github/workflows/auto-git-release.yml#L28

But I prefer doing it using github action. As for Commitizen, it's yet another opinionated tool, I don't have a strong opinion regarding best practices on this so I prefer not to integrate it by default, and make it up to people to add it on their own.

**Reply by samuelcastro** [reply]
Ok, semantic Release supports github actions as well (https://github.com/semantic-release/semantic-release/blob/master/docs/recipes/github-actions.md).

**Reply by samuelcastro** [reply]
And commitzen is all about maintainability, and I quote: "It also empowers project maintainers to create or use predefined commit message conventions in their repos to better communicate their expectations to potential contributors.".

Many (and important) open sources projects out there use it, like webpack, testing-library, etc.

But anyway, I just wanted to improved the experience as a whole, specially if you work with a big team.

**Reply by Vadorequest** [reply]
Semantic versioning is for apps published onto NPM, that's not the case of a Next.js app, it's unrelated AFAICT.

Updating the package.json:version isn't the way I'd go for a Next.js project, I've tried that way and it's not a good experience. 

At the moment, NRN versions use a semantic versioning system based on the commit messages and are automatically published as Git tags and releases upon commit pushes. I don't plan on adding anything more to that. (and I don't see what would be necessary)

**Reply by samuelcastro** [reply]
> Semantic versioning is for apps published onto NPM

For applications as well.

**Reply by Vadorequest** [reply]
I meant the Semantic Versioning **package** is for NPM apps, the "semantic versioning" **concept** applies to any kind of app indeed, you're right.
